pkg/sql2pgroll/alter_table.go

```diff
@@ -158,11 +161,27 @@ func convertAlterTableAddUniqueConstraint(stmt *pgq.AlterTableStmt, constraint *
 	}, nil
 }
 
-// convertAlterTableDropColumn converts SQL statements like:
+// convertAlterTableSetColumnDefault converts SQL statements like:
 //
-// `ALTER TABLE foo DROP COLUMN bar
+// `ALTER TABLE foo COLUMN bar SET DEFAULT 'foo'`
+// `ALTER TABLE foo COLUMN bar SET DEFAULT null`
+// `ALTER TABLE foo COLUMN bar DROP DEFAULT`
 //
-// to an OpDropColumn operation.
+// to an OpAlterColumn operation.
+func convertAlterTableSetColumnDefault(stmt *pgq.AlterTableStmt, cmd *pgq.AlterTableCmd) (migrations.Operation, error) {
```

Comment:
Since `OpAlterColumn` only supports nullable string defaults, this will fail for any other kind of default for example non strings or the value of a function call.

In those cases should we fall back to raw sql?

Reply:
It looks like if the new default value is a constant, we can support these types:

```go
type A_Const struct {
	state         protoimpl.MessageState
	sizeCache     protoimpl.SizeCache
	unknownFields protoimpl.UnknownFields

	// Types that are assignable to Val:
	//
	//	*A_Const_Ival
	//	*A_Const_Fval
	//	*A_Const_Boolval
	//	*A_Const_Sval
	//	*A_Const_Bsval
	Val      isA_Const_Val `protobuf_oneof:"val"`
	Isnull   bool          `protobuf:"varint,10,opt,name=isnull,proto3" json:"isnull,omitempty"`
	Location int32         `protobuf:"varint,11,opt,name=location,proto3" json:"location,omitempty"`
}
```

And then convert them to a string

Reply:
I've done this